### ISSUE TITLE
Add support for matching patterns along the string

### ISSUE THREAD

**Issue by kevinokerlund** [RegularWords/js-regular-words#5]
This is a really rough idea at the moment, but it should support things like:
```
match 3 numbers
then 3 letters
maybe a: #
etc
```

**Comment by kevinokerlund** [comment]
@mikkelrd, what do you think about these for this issue?
```javascript
let phone = `
	start of string
	maybe: +
	match: 1-
	match 3 digits
	match any of these: -,.
	match 3 digits
	match any of these: -,.
	match 4 digits
	end of string
`;
```

**Comment by kevinokerlund** [comment]
Resolved in PR #7
